XouDevSpec/Classes/features/MangaScreen/ViewModel/MangaViewModel.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  MangaViewModel.swift
+//  XouDevSpec
+//
+//  Created by Xander Schoeman on 2020/04/22.
+//
+
+import Foundation
+
+public protocol MangaViewModelProtocol: class {
+    func getMangaData(searchString: String, genreString: String)
+    
+}
+
+public class MangaViewModel {
+    var manga = [MangaDetails]()
+    weak var view: MangaViewProtocol?
+    var Repo: ApiJikanCallerProtocol
+    
+    public init(with view: MangaViewProtocol, repo: ApiJikanCallerProtocol) {
+        self.view = view
+        self.Repo = repo
+    }
+    
+     
+}
+
+extension MangaViewModel: MangaViewModelProtocol {
+
+
+    public func getMangaData(searchString: String, genreString: String){
+        Repo.initStrings(search: searchString, genre: genreString)
+        Repo.getMangaData { [weak self] result in
+            switch result {
+            case .failure( let error):
+                print(error)
+            case .success(let animes):
+                self?.manga = animes
```

Comment:
Lol also naming issue but I would rename this as `let manga` so its less confusing

Reply:
Just also remember to change it in the MangaViewModelProtocol

Reply:
That would make more sense..... I re used the code from the anime side and thought I changes everything so that it makes sense in the manga side. Seems I missed a few places. Nice find. I'll change it.